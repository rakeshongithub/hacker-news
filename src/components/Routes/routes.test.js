import React, { Suspense } from "react";
import { shallow } from "enzyme";
import { create } from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

import AppRoutes, { TopNewsLists, NewNewsLists } from "./routes";

describe("App Component", () => {
  it("renders app routes component without crashing", () => {
    const wrapper = shallow(<AppRoutes />);
    expect(wrapper.exists()).toBe(true);
  });

  it("renders lazy components", async () => {
    const root = create(
      <div className="containt-wrapper">
        <Suspense fallback={<div>loading...</div>}>
          <BrowserRouter>
            <AppRoutes />
          </BrowserRouter>
        </Suspense>
      </div>
    );

    await TopNewsLists;
    await NewNewsLists;
    expect(root).toMatchSnapshot();
  });
});
