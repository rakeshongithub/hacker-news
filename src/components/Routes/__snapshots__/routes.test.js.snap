// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders lazy components 1`] = `
<div
  className="containt-wrapper"
>
  <div
    className="containt-wrapper"
  >
    <div>
      loading...
    </div>
  </div>
</div>
`;
